velox/docs/develop/window.rst

```diff
@@ -0,0 +1,202 @@
+========================
+Window Operator
+========================
+
+Velox supports window function evaluation using Window operator. In this guide
+we will discuss some intricate design issues in this operator.
+
+
+Window frames
+-------------
+Window functions can optionally include a FRAME clause. The FRAME clause
+can be thought of as a spec for a sliding window of rows which bound
+the window function computation for the given row.
+
+Not all window functions are bound by the FRAME clause.
+
+- Aggregates computed as window functions and value functions :func:`first_value`,
+  :func:`last_value` and :func:`nth_value` honor window frames.
+- Rank functions :func:`row_number`, :func:`rank`, :func:`dense_rank`,
+  :func:`percent_rank`, :func:`ntile`, :func:`cume_dist` and value functions
+  :func:`lead` and :func:`lag` are not affected by window frames.
+
+A frame can be ROWS type or RANGE type, and it runs from frame_start to
+frame_end. A FRAME clause is one of
+
+.. code-block::
+
+   {RANGE|ROWS} frame_start
+   {RANGE|ROWS} BETWEEN frame_start AND frame_end
+
+frame_start and frame_end can be any of:
+
+.. code-block::
+
+   UNBOUNDED PRECEDING
+   expression PRECEDING
+   CURRENT ROW
+   expression FOLLOWING
+   UNBOUNDED FOLLOWING
+
+**ROWS mode**
```

Comment:
This and following 2 sections seem to be repeating the information found in https://facebookincubator.github.io/velox/functions/presto/window.html. Would it make sense to enhance https://facebookincubator.github.io/velox/functions/presto/window.html instead?

Reply:
There is a bit of repetition but the context are entirely different. The functions doc is more declarative is saying what the frame means for the function. The develop doc is explaining what those clauses mean for the frame indices. The sentences are framed in that way.

I have now added a link from the functions doc to the window frame doc.